src/Raven.Server/Monitoring/Snmp/SnmpWatcher.cs

```diff
@@ -40,22 +40,57 @@ public class SnmpWatcher
         public SnmpWatcher(RavenServer server)
         {
             _server = server;
+            _server.ServerStore.LicenseManager.LicenseChanged += OnLicenseChanged;
         }
 
-        public void Execute()
+        private void OnLicenseChanged()
         {
             if (_server.Configuration.Monitoring.Snmp.Enabled == false)
                 return;
 
-            if (_server.ServerStore.LicenseManager.CanUseSnmpMonitoring() == false)
+            _locker.Wait();
+
+            try
+            {
+                var snmpEngine = _snmpEngine;
+                if (snmpEngine == null) // precaution
+                    return;
+
+                var activate = _server.ServerStore.LicenseManager.CanUseSnmpMonitoring();
```

Comment:
What happens when the license was updated/replaced?
Will it start the SNMP twice?

Reply:
I've checked the implementation of snmpEngine and if the engine is started already this is a no-op